src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -68,101 +70,83 @@ g.test('visibility_and_dynamic_offsets')
     }, !success);
   });
 
-g.test('min_buffer_binding_size')
+g.test('non_required_parameter_combined_with_type_separately')
   .params(
     params()
       .combine(poptions('type', kBindingTypes))
-      .combine(poptions('minBufferBindingSize', [undefined, 0, 4]))
+      .combine([
+        ...poptions('minBufferBindingSize', [undefined, 0, 4]),
+        ...poptions('textureComponentType', [undefined, ...kTextureComponentTypes]),
+        ...poptions('multisampled', [undefined, false, true]),
+        ...poptions('viewDimension', [undefined, ...kTextureViewDimensions]),
+        ...poptions('storageTextureFormat', [undefined, ...kTextureFormats]),
+      ])
   )
-  .fn(async t => {
-    const { type, minBufferBindingSize } = t.params;
+  .fn(t => {
+    const {
+      type,
+      minBufferBindingSize,
+      textureComponentType,
+      multisampled,
+      viewDimension,
+      storageTextureFormat,
+    } = t.params;
 
     let success = false;
     if (
-      minBufferBindingSize === undefined ||
-      minBufferBindingSize === 0 ||
-      type in kBufferBindingTypeInfo
+      (minBufferBindingSize === undefined ||
+        minBufferBindingSize === 0 ||
+        type in kBufferBindingTypeInfo) &&
+      (textureComponentType === undefined || kBindingTypeInfo[type].resource === 'sampledTex') &&
+      (multisampled === false || kBindingTypeInfo[type].resource === 'sampledTex') &&
+      (viewDimension === undefined || type in kTextureBindingTypeInfo) &&
+      (storageTextureFormat === undefined ||
+        (kBindingTypeInfo[type].resource === 'storageTex' &&
+          kTextureFormatInfo[storageTextureFormat].storage))
```

Comment:
nit: this is a bit hard to read. prefer to split each requirement into it's own boolean, 
then we can have:
```typescript
const success = minBufferSizeForBuffers && textureComponentTypeForSampledTex && ...
```

Reply:
Done.